corehq/blobs/mixin.py

```diff
@@ -255,9 +259,19 @@ class BlobHelper(object):
     """
 
     def __init__(self, doc, database):
-        if doc.get("_id") is None:
+        from couchforms.models import all_known_formlike_doc_types
+
+        if doc['doc_type'] in all_known_formlike_doc_types():
+            # XFormError might have its id changed, but the blob bucket id
+            #   will still be orig_id
+            bucket_id = doc.get('orig_id') or doc['_id']
+        else:
+            bucket_id = doc['_id']
+
+        if bucket_id is None:
             raise TypeError("BlobHelper requires a real _id")
-        self._id = doc["_id"]
+
+        self._id = bucket_id
```

Comment:
Should also add
```py
self.blobdb_bucket_id = bucket_id
```
here because lines like 311 call `BlobMixin` methods passing an instance of `BlobHelper` as first argument:
```py
BlobMixin.put_attachment(self, content, name, *args, **kw)
```
This is possible because those methods use the [`document_method` decorator](https://github.com/dimagi/commcare-hq/blob/11edc1be91610c7e79ab0b297ee4bdcd2e14e054/corehq/blobs/util.py#L32).